ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -903,18 +913,39 @@ long getAllocatedUsableCapacity() {
     private final boolean isAggregatedViewHandler;
 
     /**
-     * @param dcOrClusterName the name of data center or the cluster this handler is associated with.
+     * @param dcName the name of data center or the cluster this handler is associated with.
+     * @param helixClusterName the name of helix cluster which hosts the Ambry topology information.
      * @param onInitializationFailure callback to be called if initialization fails in a listener call.
      * @param isAggregatedViewHandler indicates if this clusterChangeHandler is a listener for changes in the entire
-     *                                cluster via helix aggregated view.
      */
-    HelixClusterChangeHandler(String dcOrClusterName, Consumer<Exception> onInitializationFailure,
+    HelixClusterChangeHandler(String dcName, String helixClusterName, Consumer<Exception> onInitializationFailure,
         boolean isAggregatedViewHandler) {
-      this.dcOrClusterName = dcOrClusterName;
+      this.dcName = dcName;
+      this.helixClusterName = helixClusterName;
       this.onInitializationFailure = onInitializationFailure;
       this.isAggregatedViewHandler = isAggregatedViewHandler;
     }
 
+    @Override
+    public void onDataNodeConfigChange(Iterable<DataNodeConfig> configs) {
+      handleDataNodeConfigChange(configs, dcName);
+    }
+
+    @Override
+    public void onIdealStateChange(List<IdealState> idealState, NotificationContext changeContext) {
```

Comment:
we should probably do the same thing for regular helix cluster change handler to **NOT** listen on the IdealState change. It doesn't really provide any information. 

IdealState actually can be used to cross verify the DataNodeConfig, but we are not doing it anywhere.

Reply:
For regular helix cluster change handler, we are currently using RoutingTableProvider instantiated using Current states. For such RoutingTableProvider, list of external views seem to be empty. Due to that, we seem to be relying on IDEAL STATES for constructing the partition-to-resource mapping.

Helix API documentation:
` 
/**
   * Returns a Collection of latest snapshot of ExternalViews. Note that if the RoutingTable is
   * instantiated using CurrentStates, this Collection will be empty.
   * @return
   */
  public Collection<ExternalView> getExternalViews() {
    return _routingTable.getExternalViews();
  }
`